packages/nimble-components/src/table/index.ts

```diff
@@ -268,19 +263,58 @@ export class Table<
         );
     }
 
+    private updateTanStack(): void {
+        const updatedOptions: Partial<TanStackTableOptionsResolved<TData>> = {
+            state: {}
+        };
+
+        if (this.updateTracker.updateColumnSort) {
+            updatedOptions.state!.sorting = this.calculateTanStackSortState();
+        }
+        if (this.updateTracker.updateColumnDefinition) {
+            updatedOptions.columns = this.calculateTanStackColumns();
+        }
+        if (this.updateTracker.updateRowIds) {
+            updatedOptions.getRowId = this.calculateTanStackRowIdFunction();
+        }
+        if (this.updateTracker.requiresTanStackDataReset) {
+            // Perform a shallow copy of the data to trigger tanstack to regenerate the row models and columns.
+            updatedOptions.data = [...this.table.options.data];
+        }
+
+        this.updateTableOptions(updatedOptions);
+    }
+
+    private updateActionMenuSlots(): void {
+        const slots = new Set<string>();
+        for (const column of this.columns) {
+            if (column.actionMenuSlot) {
+                slots.add(column.actionMenuSlot);
+            }
+        }
+        this.actionMenuSlots = Array.from(slots);
+    }
+
+    private validate(): void {
+        this.validateColumnIds();
+        this.validateColumnSortIndices();
+        this.tableValidator.validateRecordIds(
+            this.table.options.data,
+            this.idFieldName
+        );
+
+        this.canRenderRows = this.checkValidity();
+    }
+
     private setTableData(newData: readonly TData[]): void {
         const data = newData.map(record => {
             return { ...record };
         });
         this.tableValidator.validateRecordIds(data, this.idFieldName);
         this.canRenderRows = this.checkValidity();
```

Comment:
I actually had originally implemented it with `data` passed as a parameter to `validate`, but I changed it based on a comment made by @atmgrifter00 pointing out that `validate` was only called with `this.table.options.data`.

I don't have a strong preference either way. Do either of you have a strong preference? @rajsite, @atmgrifter00

Reply:
Thing I wanted to avoid is having multiple different validation routines.

Maybe it's not true if you trace the code but feels like you can end up in a situation right now where depending on the order of operations of manipulating properties you maybe could end up with different issues reported by table validity.

By always calling the same validation function it feels less likely to be a possible issue.

What about:

```ts
validate() {
    this.validateWithData(this.table.options.data);
}
validateWithData (data) {...}
```

Reply:
That seems like a reasonable compromise to me.

Reply:
Reopening, the intention here was to replace doing just: this.tableValidator.validateRecordIds(data, this.idFieldName);
with the new validateWithData